src/ledger/grainAllocation.js

```diff
@@ -225,25 +220,18 @@ function computeBalancedReceipts(
     }
   }
 
-  let totalPaid = G.ZERO;
-  const receipts = mapToArray(userUnderpayment, ([address, underpayment]) => {
-    const underpaymentProportion = G.toFloatRatio(
-      underpayment,
-      totalUnderpayment
-    );
-    const amount = G.multiplyFloat(budget, underpaymentProportion);
-    totalPaid = G.add(totalPaid, amount);
-    return {
-      address,
-      amount,
-    };
-  });
-  if (G.gt(totalPaid, budget)) {
-    throw new Error(
-      `invariant violation: paid ${totalPaid} greater than budget ${budget}`
-    );
-  }
-  return receipts;
+  const underpaymentTuples: [NodeAddressT, G.Grain][] = mapToArray(
+    userUnderpayment,
+    (x) => x
+  );
+  const underpaymentFractions = underpaymentTuples.map(
+    (x) => Number(x[1]) / Number(totalUnderpayment)
+  );
+  const pieces = G.splitBudget(budget, underpaymentFractions);
```

Comment:
nit: descriptive name
```suggestion
  const grainPieces = G.splitBudget(budget, underpaymentFractions);
```

Reply:
I'm fine taking nits if they are added thru the suggestion UI (just lower friction); could you please make the suggestion apply the whole rename?